source/slang-capture-replay/replay/file-processor.cpp

```diff
@@ -0,0 +1,132 @@
+#include "file-processor.h"
+#include "../util/capture-utility.h"
+#include "parameter-decoder.h"
+
+namespace SlangCapture
+{
+    FileProcessor::FileProcessor(const std::string& filename)
+    {
+        Slang::String path(filename.c_str());
+        Slang::FileMode fileMode = Slang::FileMode::Open;
+        Slang::FileAccess fileAccess = Slang::FileAccess::Read;
+        Slang::FileShare fileShare = Slang::FileShare::None;
+
+        // Open the capture file with read-only access
+        SlangResult res = m_inputStream.init(path, fileMode, fileAccess, fileShare);
+
+        if (res != SLANG_OK)
+        {
+            SlangCapture::slangCaptureLog(SlangCapture::LogLevel::Error, "Failed to open file %s\n", filename.c_str());
+            std::abort();
```

Comment:
There are many ways to handle an error case. Killing the entire process is probably the last thing we should do.

Reply:
I cannot think of we can do here if we're failing to open the file?

Reply:
Right, you cannot do anything in the constructor.
I like to suggest that you make a dedicated method for opening a file like
```
bool OpenFile(const std::string &filename);
```